tests/sentry/utils/kvstore/test_bigtable.py

```diff
@@ -0,0 +1,156 @@
+import itertools
+import os
+from typing import Iterator, Optional
+
+import pytest
+from google.oauth2.credentials import Credentials
+from sentry.utils.kvstore.abstract import K, KVStorage, V
+from sentry.utils.kvstore.bigtable import BigtableKVStorage
+
+
+@pytest.fixture
+def credentials() -> Credentials:
+    if "BIGTABLE_EMULATOR_HOST" not in os.environ:
+        pytest.skip(
+            "Bigtable is not available, set BIGTABLE_EMULATOR_HOST enironment variable to enable"
+        )
+
+    # The bigtable emulator requires _something_ to be passed as credentials,
+    # even if they're totally bogus ones.
+    return Credentials.from_authorized_user_info(
+        {key: "invalid" for key in ["client_id", "refresh_token", "client_secret"]}
+    )
+
+
+@pytest.mark.parametrize(
+    "compression,flag,expected_prefix",
+    [
+        (None, None, b"{"),
+        ("zlib", BigtableKVStorage.Flags.COMPRESSED_ZLIB, (b"\x78\x01", b"\x78\x9c", b"\x78\xda")),
+        ("zstd", BigtableKVStorage.Flags.COMPRESSED_ZSTD, b"\x28\xb5\x2f\xfd"),
+    ],
+    ids=["zlib", "ident", "zstd"],
+)
+def test_compression_raw_values(
+    compression: Optional[str],
+    flag: BigtableKVStorage.Flags,
+    expected_prefix: bytes,
+    credentials: Credentials,
+    request,
+) -> None:
+    store = BigtableKVStorage(
+        project="test",
+        compression=compression,
+        client_options={"credentials": credentials},
+    )
+
+    store.bootstrap()
+    request.addfinalizer(store.destroy)
```

Comment:
This is clumsy but I wanted to avoid mutating `store.compression` from the outside.

Reply:
what I usually do is to return a factory function from the fixture if I need custom args passed

```
@pytest.fixture
def store(request):
    def inner(**kwargs):
        rv = BigtableKVStorage(project='test', client_options={...}, **kwargs)
        request.addfinalizer(rv.destroy)
        return rv

	return inner
```

this makes you write `store = store()` in all tests, or you can have two fixtures `create_store` and `store`, the latter of which is a shortcut fixture